File: Rookies_EcommerceWebsite.Customer/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rookies_EcommerceWebsite.Customer.Models
{
    public class Product
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Name { get; set; }
        public string Description { get; set; }
        public string[] Images { get; set; }
        public string Slug { get; set; }
        public ulong? Price { get; set; }
        public bool IsDeleted { get; set; } = false;
        public DateTime? CreatedDate { get; set; } = DateTime.Now;
        public DateTime? UpdatedDate { get; set; } = DateTime.Now;
        public ICollection<Variant> Variants { get; set; } = null!;
        public ICollection<Rating> Ratings { get; set; } = null!;
        public string CategoryId { get; set; }
        //public Category Category { get; set; } = null!;
    }
}

File: admin/src/types/Product.ts
export type Product = {
    id: string,
    name: string
    description: string
    images: string[]
    slug: string
    price: number
    categoryId: string
    created_date: string
}

export type ProductCreateDto = {
    name: string
    description: string
    images?: string[]
    slug: string
    price: number
    categoryId: string
    variants: Variant[]
}

export type ProductUpdateDto = {
    name: string
    description: string
    images?: string[]
    slug: string
    price: number
    categoryId: string
}

export type ProductFormType = {
    name: string
    description: string
    images?: FileList
    slug: string
    price: number
    category: string
    variants: Variant[]
}

export type Variant = {
    name: string
    stock: number
}